tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,601 @@
+
+    expect(root.find('.UserProfileEdit-submit-button'))
+      .toHaveProp('disabled', false);
+  });
+
+  it('renders a Not Found page when logged user cannot edit another user', () => {
+    const username = 'current-logged-user';
+    const { store } = dispatchSignInActions({
+      userProps: {
+        ...defaultUserProps,
+        username,
+        permissions: [],
+      },
+    });
+
+    // Create a user with another username.
+    const user = createUserAccountResponse({ username: 'willdurand' });
+    store.dispatch(loadUserAccount({ user }));
+
+    // Try to edit this user with another username.
+    const params = { username: user.username };
+    const root = renderUserProfileEdit({ params, store });
+
+    expect(root.find(NotFound)).toHaveLength(1);
+  });
+
+  it('allows to edit another user if logged user has USERS_EDIT permission', () => {
+    const username = 'current-logged-user';
+    const { store } = dispatchSignInActions({
+      userProps: {
+        ...defaultUserProps,
+        username,
+        permissions: [USERS_EDIT],
+      },
+    });
+
+    // Create a user with another username.
+    const user = createUserAccountResponse({ username: 'willdurand' });
+    store.dispatch(loadUserAccount({ user }));
+
+    // Try to edit this user with another username.
+    const params = { username: user.username };
+    const root = renderUserProfileEdit({ params, store });
+
+    expect(root.find(NotFound)).toHaveLength(0);
+    expect(root.find('.UserProfileEdit-username'))
+      .toHaveProp('value', user.username);
+
+    const linkItems = root.find('.UserProfileEdit-user-links li');
+    expect(linkItems.at(0).find(Link))
+      .toHaveProp('children', "View user's profile");
+    expect(linkItems.at(1).children()).toHaveText("Edit user's profile");
+
+    expect(root.find('.UserProfileEdit--Card').first())
+      .toHaveProp('header', 'Account for willdurand');
+
+    expect(root.find('.UserProfileEdit--help-email')).toHaveLength(0);
+
+    expect(root.find('.UserProfileEdit-aside')).toHaveText(oneLine`Tell users a
+      bit more information about this user. These fields are optional, but
+      they'll help other users get to know willdurand better.`
+    );
+
+    expect(root.find({ htmlFor: 'biography' }))
+      .toHaveText('Introduce willdurand to the community');
+  });
+
+  it('renders errors', () => {
+    const { store } = dispatchSignInActions();
+    const errorHandler = new ErrorHandler({
+      id: 'some-id',
+      dispatch: store.dispatch,
+    });
+    errorHandler.handle(new Error('unexpected error'));
+
+    const root = renderUserProfileEdit({ errorHandler, store });
+
+    expect(root.find(ErrorList)).toHaveLength(1);
+  });
+
+  it('returns a 404 if current user is not logged in', () => {
+    const { store } = dispatchClientMetadata();
+
+    const root = renderUserProfileEdit({ store });
+
+    expect(root.find(NotFound)).toHaveLength(1);
+  });
+});
```

Comment:
I think this is missing a test case for editing a user and changing the username, because there is currently a problem with that and it didn't cause any tests to fail.

Reply:
right, I totally missed that case 😞

Reply:
I have also added test cases for when we are editing another user profile and the username has changed (the URL must be updated)